drivers/clock_control/clock_control_nrf2_lfclk.c

```diff
@@ -56,49 +55,52 @@ static const struct clock_options {
 		.precision = 1,
 		.src = NRFS_CLOCK_SRC_LFCLK_SYNTH,
 	},
-#if LFCLK_HAS_LFXO
-#if DT_ENUM_HAS_VALUE(LFCLK_LFXO_NODE, mode, crystal)
-	{
-		.accuracy = LFCLK_LFXO_ACCURACY,
-		.src = NRFS_CLOCK_SRC_LFCLK_XO_PIERCE,
-	},
-	{
-		.accuracy = LFCLK_LFXO_ACCURACY,
-		.precision = 1,
-		.src = NRFS_CLOCK_SRC_LFCLK_XO_PIERCE_HP,
-	},
-#elif DT_ENUM_HAS_VALUE(LFCLK_LFXO_NODE, mode, external_sine)
-	{
-		.accuracy = LFCLK_LFXO_ACCURACY,
-		.precision = 0,
-		.src = NRFS_CLOCK_SRC_LFCLK_XO_EXT_SINE,
-	},
-	{
-		.accuracy = LFCLK_LFXO_ACCURACY,
-		.precision = 1,
-		.src = NRFS_CLOCK_SRC_LFCLK_XO_EXT_SINE_HP,
-	},
-#elif DT_ENUM_HAS_VALUE(LFCLK_LFXO_NODE, mode, external_square)
-	{
-		.accuracy = LFCLK_LFXO_ACCURACY,
-		.precision = 0,
-		.src = NRFS_CLOCK_SRC_LFCLK_XO_EXT_SQUARE,
-	},
-#else
-#error "unsupported LFXO mode"
-#endif
-#endif
 };
 
 struct lfclk_dev_data {
 	STRUCT_CLOCK_CONFIG(lfclk, ARRAY_SIZE(clock_options)) clk_cfg;
 	struct k_timer timer;
+	uint16_t max_accuracy;
 };
 
 struct lfclk_dev_config {
 	uint32_t fixed_frequency;
 };
 
+static int lfosc_get_accuracy(uint16_t *accuracy)
```

Comment:
It would be better to place this function next to `lfclk_init()` which is its only user.